sdk/src/types/api/core/response.rs

```diff
@@ -117,20 +118,62 @@ pub struct SubmitBlockResponse {
     pub block_id: BlockId,
 }
 
-/// Describes the ledger inclusion state of a transaction.
+/// Describes the state of a block.
 #[derive(Clone, Copy, Debug, Eq, PartialEq)]
 #[cfg_attr(
     feature = "serde",
     derive(serde::Serialize, serde::Deserialize),
     serde(rename_all = "camelCase")
 )]
-pub enum LedgerInclusionState {
-    Conflicting,
-    Included,
-    NoTransaction,
+pub enum BlockState {
+    // Stored but not confirmed or contains not yet included transaction.
+    Pending,
+    // Confirmed with the first level of knowledge.
+    Confirmed,
+    // Included and can no longer be reverted.
+    Finalized,
+    // Rejected by the node, and user should reissue payload if it contains one.
+    Rejected,
+    // Not successfully issued due to failure reason.
+    Failed,
 }
 
-/// Response of GET /api/core/v3/blocks/{block_id}/metadata.
+/// Describes the state of a transaction.
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+#[cfg_attr(
+    feature = "serde",
+    derive(serde::Serialize, serde::Deserialize),
+    serde(rename_all = "camelCase")
+)]
+pub enum TransactionState {
+    // Stored but not confirmed or contains not yet included transaction.
```

Comment:
I'd mention that's it's the block that contains not yet included tx, otherwise it's a bit weird